tools/run_tests/run_tests_matrix.py

```diff
@@ -130,6 +130,15 @@ def _create_test_jobs(extra_args=[], inner_jobs=_DEFAULT_INNER_JOBS):
                              extra_args=extra_args,
                              inner_jobs=inner_jobs)
 
+  for compiler in ['python3.4', 'python3.5', 'python3.6', 'pypy']:
```

Comment:
Depending on how long each of the tests suites takes, we might not have the capacity to run full build and tests for all 4 python configuration (I remember python tests used to take much longer than tests for other languages - it might not be the case anymore).
It might make sense to run tests for python2.7 and python3.6 (oldest and newest) and only run the build for the others? Not an expert on what the differences between python versions are though.

Also, it doesn't make sense to add the same configurations to both  "basictests" and "portability" (portability does exactly the same build but it skips the tests) -  e.g. adding  python3.6 to portability once it's already in basic tests only wastes resources.